pylint/checkers/typecheck.py

```diff
@@ -1359,7 +1359,7 @@ def visit_call(self, node):
         # Analyze the list of formal parameters.
         args = list(itertools.chain(called.args.posonlyargs or (), called.args.args))
         num_mandatory_parameters = len(args) - len(called.args.defaults)
-        parameters = []
+        parameters: List[List[Any]] = []
```

Comment:
```suggestion
        parameters: List[List[Union[Tuple[Optional[str], Any], bool]]] = []
```
At some point this should probably be revisited 😅

Reply:
The problem with `Union` is that mypy complains that it can't iterate over `bool`.
Using `Tuple` is impossible Because of call to `append` (mypy complains) and `List` can't have more than 1 argument. Do you have better typing?

Reply:
No without a bigger refactoring. Let's leave it at `List[List[Any]]` for now.